test/integration/bulk-load-test.js

```diff
@@ -132,3 +133,124 @@ exports.bulkLoadError = function(test) {
   );
   connection.execSqlBatch(request);
 };
+
+// Test if bulk load honours constraints in the destination table
+exports.bulkLoad_Constraint = function(test) {
+  const connection = this.connection;
+  const bulkLoad = connection.newBulkLoad('#tmpTestTable3', function(
+    err,
+    rowCount
+  ) {
+    test.ok(
+      err,
+      'An error should have been thrown to indicate the conflict with the CHECK constraint.'
+    );
+    test.done();
+  });
+  bulkLoad.addColumn('id', TYPES.Int, {
+    nullable: true
+  });
+  bulkLoad.addOptions(BULK_LOAD_OPTIONS.CHECK_CONSTRAINTS);
+  const request = new Request(
+    'CREATE TABLE #tmpTestTable3 ([id] int,  CONSTRAINT chk_id CHECK (id BETWEEN 0 and 50 ))',
+    function(err) {
+      test.ifError(err);
+      bulkLoad.addRow({
+        id: 555
+      });
+      connection.execBulkLoad(bulkLoad);
+    }
+  );
+  connection.execSqlBatch(request);
+};
+
+// Test if bulk load honours trigger dependent on destination table
+exports.bulkLoad_Triggers = function(test) {
```

Comment:
If you think it makes sense, you can give test cases more descriptive names by using the `[]` syntax:

```
exports['BulkLoad with FIRE_TRIGGERS'] = ...
```